### DISCUSSION TITLE
s3-static-website redirect_rule syntax example isn't doing it for me.

### DISCUSSION THREAD

**Discussion by CarltonHenderson** [gruntwork-io/knowledge-base#469] [Help] [answered]
Hi, I'm trying to spin up a static s3 site with some redirect rules and just can't figure out the right syntax.

The gruntworks example I'm using is [this one](https://github.com/gruntwork-io/terraform-aws-static-assets/blob/v0.15.3/modules/s3-static-website/variables.tf#L88-L99) in the variables file of the right version of the module I'm using.

```hcl
  routing_rule = {
    condition = {
      key_prefix_equals  = "docs/"
    }

    redirect = {
      hostname = "example"
      http_redirect_code = "403"
      protocol = "https"
      replace_key_prefix_with = "documents/"
    }
  }
```

However, I'm getting this error

```
╷
│ Error: Insufficient redirect blocks
│ 
│   on .terraform/modules/static_website.static_website/modules/s3-static-website/main.tf line 78, in resource "aws_s3_bucket_website_configuration" "website":
│   78:     content {
│ 
│ At least 1 "redirect" blocks are required.
╵
╷
│ Error: Insufficient redirect blocks
│ 
│   on .terraform/modules/static_website.static_website/modules/s3-static-website/main.tf line 78, in resource "aws_s3_bucket_website_configuration" "website":
│   78:     content {
│ 
│ At least 1 "redirect" blocks are required.
╵
```

I feel like I'm missing something small, but it's small so I can't find it. Sadly I also can't find a gruntworks example except the one in the commented out hcl.

Thanks,

Carl

---

<ins datetime="2022-06-15T17:11:22Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108782">Tracked in ticket #108782</a></p>
</ins>

**Comment by CarltonHenderson** [comment]
btw, shouldn't `hostname` be `host_name`? It seems it should be according to another error i saw while experimenting. Anyhow, it doesn't fix my problem above, but i wanted to mention it in the same thread so you don't have to discover it on your own
```
│   on .terraform/modules/static_website.static_website/modules/s3-static-website/main.tf line 91, in resource "aws_s3_bucket_website_configuration" "website":
│   91:           host_name               = lookup(redirect.value, "host_name", null)
```

**Reply by rhoboat** [reply]
Yes, that's right. The example in the comment has an error there. I'll make sure that gets updated. I'm able to reproduce the issue you're seeing. Methinks we have a bug in the module!

**Comment by rhoboat** [comment]
Thanks Carlton! I'm working on this and will get back to you soon!

**Comment by rhoboat** [comment, marked as answer]
I have a potential fix for this in our module code [here](https://github.com/gruntwork-io/terraform-aws-static-assets/pull/111). We're waiting for tests to pass, and also for a review to make sure we didn't miss something.

**Reply by rhoboat** [reply]
This has been fixed and [released here](https://github.com/gruntwork-io/terraform-aws-static-assets/releases/tag/v0.15.4).

**Reply by CarltonHenderson** [reply]
Thanks, that got it working for one redirect! However, multiple redirects on one bucket is still impossible. Fortunately I don't need that right now but I will soon. As a workaround I might be able to go back to an older version of this module which did suit my purposes. (v0.5.3)

**Reply by CarltonHenderson** [reply]
More comments and details on the bug fix pr https://github.com/gruntwork-io/terraform-aws-static-assets/pull/111
